### ISSUE TITLE
Split up the getMembers Handler into two Handlers, and modify Search

### ISSUE THREAD

**Issue by kofmangregory** [keepid/keepid_server#1]
Currently, the getMembers Handler in the UserController is responsible for 3 things:
1) Retrieving the members of an organization
2) Searching for members of an organization with first name and last name matching
3) Paginating the results

Server-side pagination has limitations. It requires querying the database ever time a page changes. Returning an array with all of the users to the frontend is not too expensive, and the frontend can utilize browser caching to speed up same-page visits. We should eliminate server-side caching altogether in this Handler.

Furthermore, the search functionality should be separated from the retrieval functionality. Searching for members of an organization should be implemented in a separate Handler that does only this. Also, consider changing the current Mongo query that does searching.

Note that there are two parts of searching: typing in the search query and hitting submit. While a user is typing, the frontend will send requests to the backend to receive recommendations. The backend should just send back the first 5 (or so) reasonable names.

On submit, all user info that matches the search query should be sent back, and the frontend will perform pagination.

In short, the three new Handlers should: 
1) Retrieve all the members of an organization
2) Retrieve the top 5 (or so) names in an organization that match the search query
3) Send back all user info that matches the search query in an organization
